extension/src/runners/baseRunner/BaseRunnerResultAnalyzer.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { Uri } from 'vscode';
+import { ITestItem, TestLevel } from '../../protocols';
+import { ITestResult, ITestResultDetails, TestStatus } from '../models';
+
+export abstract class BaseRunnerResultAnalyzer {
+    protected testResults: Map<string, ITestResultDetails> = new Map<string, ITestResultDetails>();
+    private readonly regex: RegExp = /@@<TestRunner-({[\s\S]*?})-TestRunner>/gm;
+
+    constructor(protected tests: ITestItem[]) {
+    }
+
+    public analyzeData(data: string): void {
+        let match: RegExpExecArray | null;
+        do {
+            match = this.regex.exec(data);
+            if (match) {
+                try {
+                    this.processData(match[1]);
+                } catch (ex) {
+                    // Swallow
+                }
+            }
+        } while (match);
+    }
+
+    public analyzeError(_error: string): void {
```

Comment:
Why start with underscore?

Reply:
Because it's not used param for now. This is to dismiss the `noUnusedParameters` check